tests/mdns/main.cpp

```diff
@@ -80,44 +82,52 @@ int Mainloop(Mdns::Publisher &aPublisher)
 void PublishSingleService(void *aContext, Mdns::State aState)
 {
     assert(aContext == &sContext);
+    uint8_t xpanid[kSizeExtPanId] = {0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48};
 
     if (aState == Mdns::kStateReady)
     {
-        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(12345, "SingleService", "_meshcop._udp.", "nn",
-                                                                      "cool", "xp", "1122334455667788", NULL));
+        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(
+                                      12345, "SingleService", "_meshcop._udp.", "nn", "cool", strlen("cool"), "xp",
+                                      reinterpret_cast<char *>(&xpanid), sizeof(xpanid), NULL));
     }
 }
 
 void PublishMultipleServices(void *aContext, Mdns::State aState)
 {
     assert(aContext == &sContext);
+    uint8_t xpanid[kSizeExtPanId] = {0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48};
 
     if (aState == Mdns::kStateReady)
     {
-        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(12345, "MultipleService1", "_meshcop._udp.", "nn",
-                                                                      "cool1", "xp", "1122334455667788", NULL));
-        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(12346, "MultipleService2", "_meshcop._udp.", "nn",
-                                                                      "cool2", "xp", "1122334455667788", NULL));
+        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(
+                                      12345, "MultipleService1", "_meshcop._udp.", "nn", "cool1", strlen("cool1"), "xp",
+                                      reinterpret_cast<char *>(&xpanid), sizeof(xpanid), NULL));
+        assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(
+                                      12345, "MultipleService2", "_meshcop._udp.", "nn", "cool2", strlen("cool1"), "xp",
+                                      reinterpret_cast<char *>(&xpanid), sizeof(xpanid), NULL));
     }
 }
 
 void PublishUpdateServices(void *aContext, Mdns::State aState)
 {
     assert(aContext == &sContext);
+    uint8_t xpanidOld[kSizeExtPanId] = {0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48};
+    uint8_t xpanidNew[kSizeExtPanId] = {0x48, 0x47, 0x46, 0x45, 0x44, 0x43, 0x42, 0x41};
 
     if (aState == Mdns::kStateReady)
     {
         if (!sContext.mUpdate)
         {
-            assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(12345, "UpdateService", "_meshcop._udp.",
-                                                                          "nn", "cool", "xp", "1122334455667788",
-                                                                          NULL));
+            assert(OTBR_ERROR_NONE == sContext.mPublisher->PublishService(
```

Comment:
nit: I know this is just a test, but generally prefer to keep actual logic outside `assert()`.